src/storage/test/StorageIndexWriteBenchmark.cpp

```diff
@@ -24,85 +26,183 @@ DEFINE_string(root_data_path, "/tmp/IndexWritePref", "Engine data path");
 namespace nebula {
 namespace storage {
 
-std::vector<storage::cpp2::Vertex> genVertices(VertexID &vId, TagID tagId) {
-    std::vector<storage::cpp2::Vertex> vertices;
+using NewVertex = nebula::storage::cpp2::NewVertex;
+using NewTag = nebula::storage::cpp2::NewTag;
+
+enum class IndexENV : uint8_t {
+    NO_INDEX       = 1,
+    ONE_INDEX      = 2,
+    MULITPLE_INDEX = 3,
+    INVALID_INDEX  = 4,
+};
+
+GraphSpaceID spaceId = 1;
+TagID tagId = 1;
+IndexID indexId = 1;
+
+std::string toVertexId(size_t vIdLen, int32_t vId) {
+    std::string id;
+    id.reserve(vIdLen);
+    id.append(reinterpret_cast<const char*>(&vId), sizeof(vId))
+        .append(vIdLen - sizeof(vId), '\0');
+    return id;
+}
+
+std::unique_ptr<kvstore::MemPartManager> memPartMan(const std::vector<PartitionID>& parts) {
+    auto memPartMan = std::make_unique<kvstore::MemPartManager>();
+    // GraphSpaceID =>  {PartitionIDs}
+    auto& partsMap = memPartMan->partsMap();
+    for (auto partId : parts) {
+        partsMap[spaceId][partId] = meta::PartHosts();
+    }
+    return memPartMan;
+}
+
+std::vector<nebula::meta::cpp2::ColumnDef> mockTagIndexColumns() {
+    std::vector<nebula::meta::cpp2::ColumnDef> cols;
+    for (int32_t i = 0; i < 3; i++) {
+        nebula::meta::cpp2::ColumnDef col;
+        col.name = folly::stringPrintf("col_%d", i);
+        col.type.type = meta::cpp2::PropertyType::INT64;
+        cols.emplace_back(std::move(col));
+    }
+    return cols;
+}
+
+std::shared_ptr<meta::NebulaSchemaProvider> mockTagSchema() {
+    std::shared_ptr<meta::NebulaSchemaProvider> schema(new meta::NebulaSchemaProvider(0));
+    for (int32_t i = 0; i < 3; i++) {
+        nebula::meta::cpp2::ColumnDef col;
+        col.name = folly::stringPrintf("col_%d", i);
+        col.type.type = meta::cpp2::PropertyType::INT64;
+        schema->addField(col.name, col.type.type);
+    }
+    return schema;
+}
+
+std::unique_ptr<meta::SchemaManager> memSchemaMan() {
+    auto schemaMan = std::make_unique<mock::AdHocSchemaManager>();
+    schemaMan->addTagSchema(spaceId, tagId, mockTagSchema());
+    return schemaMan;
+}
+
+std::unique_ptr<meta::IndexManager> memIndexMan(IndexENV type) {
+    auto indexMan = std::make_unique<mock::AdHocIndexManager>();
+    switch (type) {
+        case IndexENV::NO_INDEX :
+            break;
+        case IndexENV::ONE_INDEX : {
+            indexMan->addTagIndex(spaceId, indexId, tagId, mockTagIndexColumns());
+            break;
+        }
+        case IndexENV::INVALID_INDEX : {
+            indexMan->addTagIndex(spaceId, indexId, -1, mockTagIndexColumns());
+            break;
+        }
+        case IndexENV::MULITPLE_INDEX : {
+            indexMan->addTagIndex(spaceId, indexId, tagId, mockTagIndexColumns());
+            indexMan->addTagIndex(spaceId, indexId + 1, tagId, mockTagIndexColumns());
+            indexMan->addTagIndex(spaceId, indexId + 2, tagId, mockTagIndexColumns());
```

Comment:
I have a question:
three indexes are created here, but all three  indexes contain the same column,
In practice, such a case does not exist.  here only for test?

Reply:
> I have a question:
> three indexes are created here, but all three indexes contain the same column,
> In practice, such a case does not exist. Here only for test?

Good question, this test case used for write speed of the index.Although the structure of the index is the same, what is really written is three copies of data.